### PR TITLE
fix: Crash in baggageEncodedDictionary

### PR DESCRIPTION
**Pull request by philipphofmann**
## :scroll: Description

The baggageEncodedDictionary method sometimes crashes with a null pointer exception in CFCharacterSetAddCharactersInString. It seems mutableCopy of NSCharacterSet.alphanumericCharacterSet or when calling stringByAddingPercentEncodingWithAllowedCharacters sometimes doesn't work. Instead of trying to fix the underlying issue of a null pointer exception, we can convert the code to Swift, which is designed for safety to avoid such mistakes. This should prevent the crash.

## :bulb: Motivation and Context

This is a frequent crash in our [internal SDK crash detection](https://sentry.sentry.io/issues/4688158870/?project=4505469596663808&query=is%3Aunresolved&referrer=issue-stream&statsPeriod=14d&stream_index=6).

## :green_heart: How did you test it?
Unit tests.

## :pencil: Checklist

You have to check all boxes before merging:

- [x] I reviewed the submitted code.
- [x] I added tests to verify the changes.
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [x] I updated the docs if needed.
- [x] Review from the native team if needed.
- [x] No breaking change or entry added to the changelog.
- [x] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

File: Sources/Swift/Helper/SentrySwiftSerialization.swift [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        var currentSize = 0
+        
+        for (key, value) in dictionary {
+            let keyDescription = key.addingPercentEncoding(withAllowedCharacters: allowedSet) ?? ""
+            let valueDescription = value.addingPercentEncoding(withAllowedCharacters: allowedSet) ?? ""
+            let item = "\(keyDescription)=\(valueDescription)"
+            if item.count + currentSize <= SENTRY_BAGGAGE_MAX_SIZE {
+                currentSize += item.count + 1 // +1 is to account for the comma that will be added for each extra item
+                items.append(item)
+            }
```
Comment (brustolin) [inline Sources/Swift/Helper/SentrySwiftSerialization.swift:23]: h: An empty key or value should not be allowed, there is not added value.

**Review by brustolin — COMMENTED** [review]
Great! I believe converting things is even better than trying to fix objc. There is one thing to talk about tho.

File: Sources/Swift/Helper/SentrySwiftSerialization.swift (hunk 1 above)
Comment (philipphofmann) [reply Sources/Swift/Helper/SentrySwiftSerialization.swift:23]: Good catch. A test should fail for this 😉

Comment (brustolin) [reply Sources/Swift/Helper/SentrySwiftSerialization.swift:23]: Yes, I believe we should add it. 
Maybe the previous understanding was that those methods don't return nil, and that is the cause of the error.

Comment (philipphofmann) [reply Sources/Swift/Helper/SentrySwiftSerialization.swift]: That makes sense. I update the code.

File: Sources/Swift/Helper/SentryBaggageSerialization.swift [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        allowedSet.insert(charactersIn: "-_.")
+        var currentSize = 0
+        
+        for (key, value) in dictionary {
+            guard let keyDescription = key.addingPercentEncoding(withAllowedCharacters: allowedSet), !keyDescription.isEmpty else {
+                continue
+            }
+            guard let valueDescription = value.addingPercentEncoding(withAllowedCharacters: allowedSet), !valueDescription.isEmpty else {
+                continue
+            }
```
Comment (brustolin) [inline Sources/Swift/Helper/SentryBaggageSerialization.swift:22]: l: 
```suggestion
            guard let keyDescription = key.addingPercentEncoding(withAllowedCharacters: allowedSet), 
            let valueDescription = value.addingPercentEncoding(withAllowedCharacters: allowedSet), !keyDescription.isEmpty && !valueDescription.isEmpty  else {
                continue
            }
```

**Review by brustolin — APPROVED** [review]
LGTM

**Pull request merged**